cf/test/test_Field.py

```diff
@@ -743,36 +743,32 @@ def test_Field_cumsum(self):
         self.assertIsNone(g.cumsum(2, inplace=True))
         self.assertTrue(g.equals(h, verbose=2))
 
+        # Check that a new cell method that has been added
+        cell_methods = h.cell_methods(todict=True)
+        self.assertEqual(len(cell_methods), len(f.cell_methods()) + 1)
+        _, cm = cell_methods.popitem()
+        self.assertEqual(cm.method, "sum")
+        self.assertEqual(cm.axes, (h.get_data_axes()[2],))
+
+        # Check the dimension coordinate bounds
+        fx = f.dimension_coordinate("X")
+        hx = h.dimension_coordinate("X")
+        self.assertTrue((hx.lower_bounds == fx.lower_bounds[0]).all())
+        self.assertTrue((hx.upper_bounds == fx.upper_bounds).all())
```

Comment:
Ah ha. That's because the cumsum always sums from the first element. So in the result, element 2 along the axis is the sum of elements 0, 1, and 2. Thus, the bounds need to span the first three elements in this case, and the lower bound is always that of the first element.

This has got me thinking as to what happens when the coordinate are decreasing in index space. It's a fact that the `lower_bounds` is always lower in the numerical sense, so for decreasing coordiantes, the lower bounds is on the "right", not the left. I'll check that things are OK in this case, and make sure this case is tested.

Reply:
Thanks for clarifying, I see now!

Reply:
The decreasing case is OK, as `Field.cumsum` works in index space (`bounds[:, 0] = bounds[0, 0]`). Nonetheless, I've added a unit test for this (https://github.com/NCAS-CMS/cf-python/pull/461/commits/99b54701f2a3e0f477ede0c096fc4073482eadd4)

Reply:
Brilliant, thanks. Marking as resolved.